File: 94_BinaryTreeInorderTraversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def inorderTraversal(self, node: TreeNode) -> List[int]:
        tree = []
        def closureTraversal(root):
            if not root:
                return tree
            if root.left:
                closureTraversal(root.left)
            tree.append(root.val)
            if root.right:
                closureTraversal(root.right)
            return tree
        return closureTraversal(node)

File: kotlin/94_BinaryTreeInorderTraversal.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    val nums: MutableList<Int> = mutableListOf<Int>()
    fun inorderTraversal(root: TreeNode?): List<Int> {
        if (root == null) { return nums }
        
        inorderTraversal(root?.left)
        nums.add(root?.`val`)
        inorderTraversal(root?.right)
        
        return nums
    }
}
